src/dune_rules/merlin.ml

```diff
@@ -47,36 +47,37 @@ module Processed = struct
     (* Failing to load the file at that point means that the configuration file
        has been written by a version of Dune in which the [Merlin.Processed.t]
        type is different from the one in the current version. *)
-    Option.map ~f:Result.return (Persist.load f)
-    |> Option.value
-         ~default:
-           (Error
-              "The current Merlin configuration has been generated by another, \
-               incompatible, version of Dune. Please rebuild the project. \
-               (Using the same version of Dune as the one running the \
-               `ocaml-merlin` server.)")
+    match Persist.load f with
+    | Some s -> Ok s
+    | None ->
+      Error
+        "The current Merlin configuration has been generated by another, \
+         incompatible, version of Dune. Please rebuild the project. (Using the \
+         same version of Dune as the one running the `ocaml-merlin` server.)"
```

Comment:
I often feel like a `Option.map_value : 'a t -> default:'b -> f:('a -> 'b) -> 'b` function would be useful.
Would that be a welcome addition to Dune's std ?

Reply:
Possibly, but in practice this function ends up being less useful since it eagerly allocates the default value. One can also have a fold:

```
val fold : 'a t -> init:'acc -> f:('a -> 'acc -> 'acc) -> 'acc
```

Which is a bit more general, but can be used to implement `map_value` without too much difficulty.